core/kerberos/kerberos-core/src/main/java/com/thinkbiganalytics/kerberos/KerberosUtil.java

```diff
@@ -20,45 +20,65 @@
  * #L%
  */
 
+import com.google.common.base.Throwables;
+
 import org.apache.hadoop.security.UserGroupInformation;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.security.PrivilegedExceptionAction;
 import java.sql.Connection;
 import java.sql.SQLException;
+import java.util.concurrent.Callable;
 
+import javax.annotation.Nonnull;
 import javax.sql.DataSource;
 
 /**
+ * Utility methods for executing code that requires an active Kerberos ticket.
  */
 public class KerberosUtil {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)